processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -51,17 +54,87 @@ protected ListeningExecutorService delegate()
   }
 
   @Override
-  public <T> ListenableFuture<T> submit(Callable<T> tCallable)
+  public <T> ListenableFuture<T> submit(final Callable<T> tCallable)
   {
+    final Callable<T> wrappedPrioritizedCallable = new PrioritizedCallable<T>()
+    {
+      @Override
+      public int getPriority()
+      {
+        if (tCallable instanceof PrioritizedCallable) {
+          return ((PrioritizedCallable) tCallable).getPriority();
+        } else {
+          return 0;
+        }
+      }
+
+      @Override
+      public T call() throws Exception
+      {
+        final Long threadUserTime;
+        if (threadMXBean.isThreadCpuTimeEnabled()) {
+          threadUserTime = threadMXBean.getCurrentThreadUserTime();
+        } else {
+          threadUserTime = null;
+        }
+        final T retval = tCallable.call();
+        if (threadUserTime != null) {
+          emitter.emit(
+              metricBuilder
+                  .setDimension("priority", String.format("%d", getPriority()))
+                  .build("exec/cpu", threadMXBean.getCurrentThreadUserTime() - threadUserTime)
+          );
+        }
+        return retval;
+      }
+    };
     emitMetrics();
-    return delegate.submit(tCallable);
+    return delegate.submit(wrappedPrioritizedCallable);
   }
 
   @Override
-  public void execute(Runnable runnable)
+  public ListenableFuture<?> submit(final Runnable runnable)
```

Comment:
I made the two submits the ones with the bulk of the impl. the `execute` is still here, but down below and simply calls `submit`.